sentry-android-navigation/src/main/java/io/sentry/android/navigation/SentryNavigationListener.kt

```diff
@@ -0,0 +1,57 @@
+package io.sentry.android.navigation
+
+import android.os.Bundle
+import androidx.navigation.NavController
+import androidx.navigation.NavDestination
+import io.sentry.Breadcrumb
+import io.sentry.Hint
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.SentryLevel.INFO
+import io.sentry.TypeCheckHint
+import java.lang.ref.WeakReference
+
+class SentryNavigationListener(
+    private val hub: IHub = HubAdapter.getInstance()
```

Comment:
We need an empty ctor to be Java friendly as well.

Reply:
I think @JvmOverloads should do? https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-overloads/

Reply:
If it generates without any params as well, that should work, we could even use this annotation within the other ctors, IIRC we've not used this annotation yet due to likely not knowing about it, good catch.

Reply:
I think it should generate yeah, will verify it

Reply:
Yeah, it does 
![image](https://user-images.githubusercontent.com/4999776/174974064-35b2d87e-50cc-4341-b143-0fc04cd89ca6.png)

Reply:
filed https://github.com/getsentry/sentry-java/issues/2124